### ISSUE TITLE
Create user/auth-service

### ISSUE THREAD

**Issue by jvikstrom** [sjchat/sjchat#13]
Create the user/auth-service. It should use jwt tokens for authentication. It also should somehow sync the (token)secrets between services.

**Comment by Pranz** [comment]
@alanihre 
From the beginning, we thought we didn't have to store any tokens on the servers at all. Instead, we would only have signed tokens where the tokens have the expiry information. However, this creates a problem if a user log outs. If a user log outs, any tokens created from the user should be invalidated.

So we're thinking about what's a good way to store the tokens on the server. To have it easily scalable and distributable, the database is the only solution, but this creates some overhead. Thoughts on this, or do you have any other suggestions on how to invalidate the tokens without storing them?

**Comment by Pranz** [comment]
@alanihre 
On another note, should the user service also be responsible for verifying tokens? We're a little unsure about the limits of responsibilities of the different services.

**Comment by alanihre** [comment]
@Pranz 
Is there any reason for why all tokens for all devices for a user should be invalidated?
When loggin out from Slack and Facebook, for example, the user is only logged out from that device and.

**Comment by alanihre** [comment]
@Pranz 
One suggestion is to put the verifying in the user service since multiple other services will get tokens from users (like rest api and web sockets services) and it would be unnecessary to have the same code at multiple places.
What do you think about that suggestion?

**Comment by Pranz** [comment]
@alanihre Yes we will go with adding authentication functions to the user service. We decided to skip a soft logout since that can be done client side (by just clearing the token cookie), but we will also do a forced logout from all devices since that could be useful when a user want to change a password or something like that.

Does that sound good?

**Comment by alanihre** [comment]
Sounds good!
How will the forced logout work?

**Comment by Pranz** [comment]
Every user has a timestamp since they last forced logout. Any token created before that timestamp is considered invalid.

**Comment by alanihre** [comment]
Sounds good!
